### PR TITLE
feat(google-locations): new location data imports in google provider

### PR DESCRIPTION
**Pull request by sevazhidkov**

### CODE REVIEW

File: providers/google/location_history_test.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	p := google{Database: provider.NewDatabase(db)}
+	require.NoError(t, p.Migrate())
+
+	require.NoError(t, p.importLocationHistoryFromFile("testdata/google/Location History/Location History.json"))
+
+	var locationHistory []LocationHistoryItem
+	require.NoError(t, db.Preload("Activities.TypeCandidates").Find(&locationHistory).Error)
+	require.Len(t, locationHistory, 4)
+
+	// Sat Nov 01 2014 14:27:19
```
Comment (shekhirin) [inline providers/google/location_history_test.go:27]: what is that.

Comment (sevazhidkov) [reply providers/google/location_history_test.go:27]: nothing you haven't seen it

Comment (shekhirin) [reply providers/google/location_history_test.go:27]: it's still there.

Comment (sevazhidkov) [reply providers/google/location_history_test.go]: i don't think so

File: providers/google/semantic_location_history_test.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+		LocationLatitudeE7:          340165480,
+		LocationLocationConfidence:  55.464928,
+		LocationLongitudeE7:         -1184961984,
+		LocationName:                "Downtown Santa Monica",
+		LocationPlaceID:             "ChIJFwxnxM-kwoARewIypbH6UsU",
+		LocationSourceInfoDeviceTag: 59293564,
+		PlaceConfidence:             "MEDIUM_CONFIDENCE",
+		VisitConfidence:             75,
+		SimplifiedRawPathPoints: []PlacePathPoint{
+			{AccuracyMeters: 24, LatE7: 340470606, LngE7: -1185254605, Time: types.DateTime(time.Date(2020, 2, 11, 17, 54, 14, 505000000, time.UTC))},
```
Comment (shekhirin) [inline providers/google/semantic_location_history_test.go:60]: how's about splitting it into multiple lines as everywhere?

File: providers/google/semantic_location_history_test.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+				LocationSourceInfoDeviceTag: 59293564,
+				PlaceConfidence:             "LOW_CONFIDENCE",
+				VisitConfidence:             75,
+				OtherCandidateLocations: []CandidateLocation{
+					{LatitudeE7: 340154424, LocationConfidence: 11.792331, LongitudeE7: -1184863484, PlaceID: "ChIJj0JGzdKkwoARw0F60V1_eQE"},
+				},
+			},
+		},
+	}
+	placeVisits[0].Model = gorm.Model{}
```
Comment (shekhirin) [inline providers/google/semantic_location_history_test.go:85]: how's about introducing a helper function for resetting the `gorm.Model` field for visit and all its children recursively?

File: providers/google/semantic_location_history_test.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+	placeVisits[0].ChildVisits[0].PlaceVisit = nil
+	placeVisits[0].ChildVisits[0].OtherCandidateLocations[0].Model = gorm.Model{}
+	placeVisits[0].ChildVisits[0].OtherCandidateLocations[0].PlaceVisitID = 0
+	assert.EqualValues(t, expectedPlaceVisit, placeVisits[0])
+
+	expectedActivitySegment := ActivitySegment{
+		Activities: []ActivityTypeCandidate{
+			{ActivityType: "IN_PASSENGER_VEHICLE", Probability: 93.55993270874023},
+		},
+		ActivityType:                       "IN_PASSENGER_VEHICLE",
```
Comment (shekhirin) [inline providers/google/semantic_location_history_test.go:100]: seems like this activity is missing in our constants. How's that possible?

File: providers/google/semantic_location_history_test.go [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+	activitySegments[0].Waypoints[0].ActivitySegment = ActivitySegment{}
+	assert.EqualValues(t, expectedActivitySegment, activitySegments[0])
+
+	placeVisits = nil
+	activitySegments = nil
+	require.NoError(t, p.importSemanticLocationHistoryFromDirectory("testdata/google/Location History/Semantic Location History/"))
+	require.NoError(t, db.Find(&placeVisits).Error)
+	require.NoError(t, db.Find(&activitySegments).Error)
+	require.Len(t, placeVisits, 2)
+	require.Len(t, activitySegments, 1)
```
Comment (shekhirin) [inline providers/google/semantic_location_history_test.go:153]: how's about also comparing before and after import like here?
https://github.com/bionic-dev/bionic/blob/55d68c6833519652a94db309fdf236f20fe80e41/providers/rescuetime/activity_history_test.go#L50

File: providers/google/semantic_location_history_test.go [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+		Preload("SimplifiedRawPathPoints").
+		Preload("TransitStops").
+		Preload("Waypoints").
+		Find(&newActivitySegments).
+		Error)
+	require.Len(t, newPlaceVisits, 2)
+	assertPlaceVisit(t, placeVisits[0], newPlaceVisits[0])
+	require.Len(t, newActivitySegments, 1)
+	assertActivitySegments(t, activitySegments[0], newActivitySegments[0])
+
```
Comment (shekhirin) [inline providers/google/semantic_location_history_test.go:152]: extra newline detected

**Pull request merged**